pages/read.tsx

```diff
@@ -24,12 +23,21 @@ export const getServerSideProps = async (context) => {
 const ReadPage = ({ article }) => {
   const router = useRouter();
   const { url } = router.query;
+
+  const backButtonClickHandler = () => {
+    if (window.history.length > 1) {
+      router.back();
```

Comment:
But why `router.back()` you go back to "trang chủ" anyway. Why waste time comparing `window.history.length`? :thinking:

Reply:
from slack conservation:
```
handle vậy là đúng cách đó, hàm router.back() nó gọi window.history.back(), như thế sẽ keep đc cái vị trí scroll của trang trước, còn cách ban đầu là jump tới địa chỉ "/", cách này trigger navigation bằng cách push 1 item mới vô history, và nó ko phải là jump back nên sẽ ko có chuyện keep đc scroll position```

Reply:
Good question. First, the reason we need to use `router.back()` is to go back in history (it called `window.history.back()` under the hood). Going this way the browser will handle the scroll position for us. In the previous way, navigate to home page at "/" is just a new push to the browser's history queue, so no scroll position will be perserved.

As for the history length check: We have 2 ways of navigate to a post content page, the first one is when the user open it directly in the home page (by clicking on the post's title, or click on the "Read more" button). The second one is when the user open the post content page in a new tab.

In the first case, go back in history is the obvious way. But in the second case, there will be no history queue, as it's the new window. So we just go to the home page for that case.